From c0616786aeca6b78a9fd76216e66273693709807 Mon Sep 17 00:00:00 2001
From: Sean Cross <xobs@kosagi.com>
Date: Thu, 9 Apr 2015 13:37:34 +0800
Subject: [PATCH 23/65] ASoC: es8328: use single writes to fix resume

The ES8328 doesn't support block writes to all registers.  Specify
single rw for regmap in order to support suspend/resume.

Signed-off-by: Sean Cross <xobs@kosagi.com>
---
 sound/soc/codecs/es8328.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/sound/soc/codecs/es8328.c b/sound/soc/codecs/es8328.c
index afa6c5d..16571fd 100644
--- a/sound/soc/codecs/es8328.c
+++ b/sound/soc/codecs/es8328.c
@@ -11,6 +11,7 @@
  */
 
 #include <linux/clk.h>
+#include <linux/clk/clk-conf.h>
 #include <linux/delay.h>
 #include <linux/of_device.h>
 #include <linux/module.h>
@@ -629,6 +630,10 @@ static int es8328_resume(struct snd_soc_codec *codec)
 
 	es8328 = snd_soc_codec_get_drvdata(codec);
 
+	ret = of_clk_set_defaults(codec->dev->of_node, false);
+	if (ret)
+		dev_err(codec->dev, "unable to set clock defaults: %d\n", ret);
+
 	ret = clk_prepare_enable(es8328->clk);
 	if (ret) {
 		dev_err(codec->dev, "unable to enable clock\n");
@@ -706,6 +711,7 @@ static int es8328_remove(struct snd_soc_codec *codec)
 const struct regmap_config es8328_regmap_config = {
 	.reg_bits	= 8,
 	.val_bits	= 8,
+	.use_single_rw	= true,
 	.max_register	= ES8328_REG_MAX,
 	.cache_type	= REGCACHE_RBTREE,
 };
-- 
2.7.3

